src/components/ListItem.jsx

```diff
@@ -4,23 +4,56 @@ import React, { useState } from 'react';
 // LOCAL IMPORTS
 import './ListItem.css';
 import { updateItem } from '../api/firebase.js';
+import { getDaysBetweenDates } from '../utils/dates.js';
 
 export function ListItem({
 	name,
 	listToken,
 	dateLastPurchased,
 	itemId,
-	totalPurchases,
-	dateCreated,
 	dateNextPurchased,
+	dateCreated,
+	totalPurchases,
 }) {
-	const now = Date.now() / 1000; //current time in seconds
-
 	// SET STATES
 	const [isChecked, setIsChecked] = useState(
-		now - dateLastPurchased?.seconds < 60 * 60 * 24,
+		Date.now() / 1000 - dateLastPurchased?.seconds < 60 * 60 * 24,
 	);
 
+	//set a variable to store urgency indicator string
+	//use conditional logic to set string
+	//if today's date - last purchase date > 60
+	//else if date next purchased - today's date > 0 && < 7
+	//"soon"
+	//if date next purchased - today's date >=7 && < 30
+	//"kind of soon"
+	//if date next purchased - today's date >= 30
+	//"not soon"
+
+	function getUrgency(dateLastPurchased = undefined, dateNextPurchased) {
+		const today = new Date();
+
+		const daysToNextPurchase = getDaysBetweenDates(
```

Comment:
Convert this value to an integer either here, or wrap `Math.round()` around the calculation of the return constant in `getDaysBetweenDates` from '../utils.dates.js' 
```suggestion
		const daysToNextPurchase = Math.round(getDaysBetweenDates(...
```